Rest/Controller/ContentController.php

```diff
@@ -73,13 +75,31 @@ public function __construct(
      *
      * @return \EzSystems\RecommendationBundle\Rest\Values\ContentData
      *
-     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundExceptionif the content, version with the given id and languages or content type does not exist
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException if the content, version with the given id and languages or content type does not exist
      * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the user has no access to read content and in case of un-published content: read versions
      */
     public function getContent($contentIdList)
     {
         $contentIds = explode(',', $contentIdList);
-        $content = $this->prepareContent($contentIds);
+        $lang = $this->request->get('lang');
+
+        $criterion = $contentTypesCriterion = array_map(function ($contentId) {
+            return new Criterion\ContentId($contentId);
+        }, $contentIds);
+
+        if (!$this->request->get('hidden')) {
+            $criterion[] = new Criterion\Visibility(Criterion\Visibility::VISIBLE);
+        }
+        if ($lang) {
+            $criterion[] = new Criterion\LanguageCode($lang);
+        }
+
+        $query = new Query();
+        $query->query = new Criterion\LogicalAnd($criterion);
+
+        $contents = $this->searchService->findContent($query)->searchHits;
```

Comment:
you can also use findLocation here and rather load the content from database.

1. search will be faster
2. there is a slight chance  an update/creation of content has not been indexed by search engine yet, at least for updates this would thus be safer.